### PR TITLE
feat/enhance text highlight string

### PR DESCRIPTION
**Pull request by israelko**
## Description
extended the `Text` component `highlightString` prop, to allow it to also receive a `HighlightStringProps` object (or an array of such objects) that enables the user to handle a highlight string `onPress` event, provide a `style` per highlight string and give it a `testID` for testing.

If a the `style` prop is not provided in a `HighlightStringProps` object, the general `highlightStringStyle` prop style is used for it.

## Changelog
`Text` component `highlightString` prop now enables to handle a highlighted string `onPress` event, provide a specific `style` to each highlighted string and give it a `testID` for better testing

## Additional info
this PR emerged from this [slack thread](https://wix.slack.com/archives/C6C38SVS4/p1688642226659659?thread_ts=1688635409.468709&cid=C6C38SVS4)

### CODE REVIEW

File: src/components/text/__tests__/index.driver.spec.tsx
Comment (lidord-wix) [inline src/components/text/__tests__/index.driver.spec.tsx:1]: This file actually tests the `Text.driver` and not the Text component.
Please move those tests to a new dedicated file that tests the Text component :)

File: src/utils/textUtils.ts
Comment (lidord-wix) [inline src/utils/textUtils.ts:1]: Feel free to add some tests for the new cases in the `textUtils.spec.js` file :)

**Review by lidord-wix — COMMENTED** [review]
Looks really good! 
Thank you very much for the contribution 🙏🏽 
Please see my comments

File: src/components/text/index.tsx [hunk 1]
```diff
@@ -38,9 +56,9 @@ export type TextProps = RNTextProps &
      */
     underline?: boolean;
     /**
-     * Substring to highlight
+     * Substring to highlight. Can be a simple string of a HighlightStringProps object, or an array of the above
```
[the remark this answers is not in the document]
Comment (israelko) [reply src/components/text/index.tsx:59]: 👍

File: src/components/text/__tests__/index.driver.spec.tsx
Comment (israelko) [reply src/components/text/__tests__/index.driver.spec.tsx:1]: @lidord-wix ok. should I test it differently or it's simply a matter of putting the same tests in a separate file? (and of course, reverting to the original `index.driver.spect.tsx` file...)

File: src/utils/textUtils.ts
Comment (israelko) [reply src/utils/textUtils.ts:1]: 👍

File: src/components/text/__tests__/index.driver.spec.tsx
Comment (lidord-wix) [reply src/components/text/__tests__/index.driver.spec.tsx:1]: It looks good to me, just a matter of file separating :)

**Pull request merged**